### PR TITLE
Update Makefile to increase support

### PR DESCRIPTION
**Pull request by arewm**
These changes were pulled out of https://github.com/release-engineering/cachito/pull/305

* Local cachito dev containers can now be run from a mac
* venv properly installs packages and will be updated at every
  invocation -- https://github.com/release-engineering/cachito/issues/95

Signed-off-by: arewm <arewm@users.noreply.github.com>

### CODE REVIEW

**Comment by arewm** [comment]
@athos-ribeiro @lcarva, FYI. Here are the Makefile changes that I will no longer be committing to #305.

**Comment by lcarva** [comment]
> @athos-ribeiro @lcarva, FYI. Here are the Makefile changes that I will no longer be committing to #305.

Nice! Out of curiosity, do the changes in #313 work on a mac?

File: Makefile
Comment (athos-ribeiro) [inline Makefile:20]: Can this Python version be a variable in the top of the file? I understand why it is nice to freeze the minor version here, but this will change every 6~12 months since we try to follow fedora releases.

Cc @chmeliik (this may be relevant when moving to f32 at some point this month).

**Review by athos-ribeiro — COMMENTED** [review]
LGTM. There is a nitpick on adding a variable for the python version used.

File: Makefile
Comment (chmeliik) [reply Makefile:20]: +1 to using a variable, there are also related documentation changes in https://github.com/release-engineering/cachito/pull/305/commits/ae24045d4be33f181db4db572eb6fd06044b5d8f (part of https://github.com/release-engineering/cachito/pull/305), those should probably be here instead

Comment (arewm) [reply Makefile:20]: @chmeliik, the documentation changes are not necessarily part of this, but I can put them here. Those are just general documentation improvements 😁

Comment (chmeliik) [reply Makefile:20]: Yeah that's fair, they can stay in there then :D

Comment (arewm) [reply Makefile:20]: @athos-ribeiro, it is definitely possible to make that change. But the Makefile is not the only place that this needs to be performed since it also needs to be updated in `tox`.

**Comment by arewm** [comment]
@lcarva, I do not know if it works with `podman-compose` since I do not have it locally. I can install `docker-compose` from brew, but not `podman-compose` therefore I imagine that it is not as likely to be a common workflow. 😁 

```
am@frybread ~/gitlab/product-configs (master) $ brew install docker-compose
Warning: docker-compose 1.27.4_1 is already installed, it's just not linked
You can use `brew link docker-compose` to link this version.
am@frybread ~/gitlab/product-configs (master) $ brew install podman-compose
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
Error: No available formula or cask with the name "podman-compose".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
```

... but I will test out this PR with the other changes you made to the Makefile/compose process...I will chase that down after I get #305 wrapped up. :)

File: Makefile
Comment (athos-ribeiro) [reply Makefile:20]: Oh, this is just a nitpick so it is easier to spot and change the value for this file in the top of the file whenever we bump the python minor version. If you think it would just make the file more verbose and not add much value, we can go ahead and merge this one (just let me know!)

**Review by athos-ribeiro — APPROVED** [review]

File: Makefile
Comment (arewm) [reply Makefile]: Clarity is good... and version updated based on upstream. 😄 

... if only there was a way to make all versions toggle-able with a single variable.

**Comment by arewm** [comment]
Since @lcarva's changes in https://github.com/release-engineering/cachito/pull/313 seem to work on a mac as well, I will remove the changes to run and just include changes to the virtual environment.

File: Makefile
Comment (arewm) [reply Makefile]: @chmeliik, I moved one of the changes I made to the README there over here because of the new Make target additions!

File: Makefile
Comment (arewm) [inline Makefile:45]: I wanted to provide a `run` command while forcing a rebuild if that is something that people wanted to always call, but I don't know what the best name for it would be. Feel free to suggest other things/tell me that this is not necessary.

**Comment by arewm** [comment]
re-requesting review since I have pushed changes after approval was submitted.

**Comment by arewm** [comment]
Bump on a review for this PR and a requested merge 😁 !

File: Makefile
Comment (chmeliik) [reply Makefile:46]: Also can't think of anything great, maybe `run-from-scratch`?

File: Makefile
Comment (chmeliik) [inline Makefile:16]: Why do we need `--system-site-packages`?

File: Makefile
Comment (chmeliik) [inline Makefile:18]:
```suggestion
# Keep run target for backwards compatibility
```

File: README.md
Comment (chmeliik) [inline README.md:205]:
```suggestion
If you suspect that the images used for [`docker-compose`](#run-a-containerized-development-environment) are out of date, you can
```

File: README.md
Comment (chmeliik) [inline README.md:206]:
```suggestion
run the containerized development environment while forcing a rebuild of the images with the following
```

**Review by chmeliik — COMMENTED** [review]
One question and a few typos, otherwise looks good to me :-)

File: Makefile
Comment (arewm) [reply Makefile:46]: `run-rebuild`?

Comment (arewm) [reply Makefile:46]: nevermind, that is too close to `run-build`.  I think the issue that I had is that I am overloading `run` as a generic term for running the development environment, but I could change targets to things like `dev-build`, `dev-run`, `dev-rebuild`, `dev-rebuild-run`. If I do this, then the `run-start-over` would instead be `dev-rebuild-run`. Thoughts?

Comment (chmeliik) [reply Makefile:46]: That sounds good to me. Maybe the `dev-` prefix is not even necessary, but I have not seen that many Makefiles so I'll defer to you on that

File: Makefile
Comment (arewm) [reply Makefile:16]: I cannot remember the specific error since I did this so long ago, but if I remember, I was hitting an error installing the packages without this parameter. Adding `--system-site-packages` was the only resolution that I could find at the time... Trying to reproduce to see.

File: Makefile
Comment (arewm) [reply Makefile:46]: Fair enough. I was just thinking to namespace them, but `build` _is_ namespaced... building the dev environment since Python doesn't need building... :D

Comment (chmeliik) [reply Makefile:46]: Hmm, now that you mention it, it *could* be confusing... Maybe having the prefix is better after all :D

File: Makefile
Comment (arewm) [reply Makefile:16]: I cannot reproduce this now, so I will remove it since it seems _better_ to have full isolation.

File: README.md
Comment (lcarva) [inline README.md:210]: The target `run-start-over` doesn't exist, right?

File: Makefile
Comment (lcarva) [inline Makefile:54]: Maybe `PYTHON_TEST_SUITE` should be called something like `TOX_ARGS` because you could use this to run something like:
```
tox -e py38 -- -x --no-cov tests/test_workers/test_tasks/test_gomod.py
```

Same for `PYTHON_VERSION_TOX`. That should probably be something like `TOX_ENVLIST` ?

File: Makefile
Comment (lcarva) [reply Makefile]: We can always namespace later if needed. There's benefit in keeping the target names short IMO.

File: Makefile
Comment (chmeliik) [inline Makefile:16]: heads-up: changed in #341 (requirements.txt have to be installed before running `setup.py develop`)

Comment (arewm) [reply Makefile]: Done.

File: README.md
Comment (arewm) [reply README.md]: addressed.

File: Makefile
Comment (arewm) [reply Makefile]: I will keep them "shorter" for now. No need to prematurely optimize.

File: Makefile
Comment (arewm) [reply Makefile]: Addressed.

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Pull request merged**
